app/views/__tests__/ExportCodeInput.spec.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import {
+  NavigationScreenProp,
+  NavigationRoute,
+  NavigationState,
+  NavigationParams,
+} from 'react-navigation';
+import { render, cleanup } from '@testing-library/react-native';
+import '@testing-library/jest-native/extend-expect';
+import { fireEvent } from '@testing-library/react-native';
+import { TracingStrategy } from '../../../app/tracingStrategy';
+import { factories } from '../../../app/factories';
+
+import ExportCodeInput from '../../../app/views/Export/ExportCodeInput';
+import { TracingStrategyProvider } from '../../TracingStrategyContext';
+
+const createTestProps = (): unknown => ({
+  navigation: jest.fn(),
+  route: {
+    params: {
+      selectedAuthority: 'pc',
+    },
+  },
+});
+
+interface NavProps {
+  navigation: NavigationScreenProp<NavigationState, NavigationParams>;
+  route: NavigationRoute;
+}
+
+afterEach(cleanup);
+
+describe('<ExportCodeInputScreen />', () => {
+  beforeEach(() => {});
+  it('when there is no text input', () => {
+    const strategy = factories.tracingStrategy.build({
+      name: 'test-strategy',
+    });
+    const { getByTestId } = renderTracingStrategyProvider(strategy);
+    const button = getByTestId('next-button');
+    expect(button).toBeDisabled();
+  });
+
+  it('when there is text input but the input code is < 6 characters', () => {
+    const strategy = factories.tracingStrategy.build({
+      name: 'test-strategy',
+    });
+    const { getByTestId } = renderTracingStrategyProvider(strategy);
+    for (let i = 0; i < 5; i++) {
+      const input = getByTestId(`input${i}`);
+      fireEvent.changeText(input, '1');
+    }
+    const button = getByTestId('next-button');
+    expect(button).toBeDisabled();
+  });
+
+  it('when the input code is 6 characters in length', () => {
+    const strategy = factories.tracingStrategy.build({
+      name: 'test-strategy',
+    });
+    const { getByTestId } = renderTracingStrategyProvider(strategy);
+    for (let i = 0; i <= 5; i++) {
+      const input = getByTestId(`input${i}`);
+      fireEvent.changeText(input, '1');
+    }
+    const button = getByTestId('next-button');
+    expect(button).not.toBeDisabled();
+  });
+});
+
+const renderTracingStrategyProvider = (strategy: TracingStrategy) => {
```

Comment:
This totally works, but is a bit of a smell to me. I feel like the tests would be better Unit tests if the CodeInput was less couple to the data model and we could pass in all of the data required as a prop.  so basically decouple the notion of a `Screen` which know how to handle navigation and application IO stuff and a `CodeInput` which know how to validate code inputs from a user.

Reply:
I think I follow but just want to make sure; are you envisioning 2 separate components, an `ExportScreen` and a `CodeInput`?

Reply:
yup something like that! I wouldnt say that is necessary for this commit, but I think that would a way to make things a bit more modular.